core/services/kraken/harbor/container.py

```diff
@@ -0,0 +1,141 @@
+import asyncio
+from typing import AsyncGenerator, Dict, List, cast
+
+import psutil
+from aiodocker import Docker
+from aiodocker.containers import DockerContainer
+from commonwealth.utils.apis import StackedHTTPException
+from fastapi import status
+from loguru import logger
+
+from harbor.contexts import DockerCtx
+from harbor.exceptions import ContainerNotFound
+from harbor.models import ContainerModel, ContainerUsageModel
+
+
+class ContainerManager:
+    @staticmethod
+    async def get_raw_container_by_name(client: Docker, container_name: str) -> DockerContainer:
+        containers = await client.containers.list(filters={"name": {container_name: True}})  # type: ignore
+        if not containers:
+            raise ContainerNotFound(f"Container {container_name} not found in running containers")
+
+        return cast(DockerContainer, containers[0])
```

Comment:
Shouldn't this return a List[DockerContainers] instead of the first, as `name` is not a unique identifier, and let the caller decide if it wants the first or all?